docs/sources/migration-guide/migrating-from-single-zone-with-helm.md

```diff
@@ -0,0 +1,715 @@
+
+   rollout_operator:
+     enabled: true
+   ```
+
+   These values are actually the default, which means that removing the values `ingester.zoneAwareReplication.enabled` and `rollout_operator.enabled` is also a valid step.
+
+1. Upgrade the installation with the `helm` command using your regular command line flags.
+
+### Migrate ingesters without downtime
+
+Before starting this procedure, set up your zones according to [Configure zone aware replication for ingesters](#configure-zone-aware-replication-for-ingesters)
+
+1. Double the series limits for tenants and the ingesters.
+
+   Explanation: while new ingesters are being added, some series will start to be written to new ingesters, however the series will also exist on old ingesters, thus the series will count twice towards limits. Not updating the limits might lead to writes to be refused due to limits violation.
+
+   The `limits.max_global_series_per_user` Mimir configuration parameter has a non-zero default value of 150000. Double the default or your value by setting:
+
+   ```yaml
+   mimir:
+     structuredConfig:
+       limits:
+         max_global_series_per_user: 300000 # <-- or your value doubled
+   ```
+
+   If you have set the Mimir configuration parameter `ingester.instance_limits.max_series` via `mimir.config` or `mimir.structuredConfig` or via runtime overrides, double it for the duration of the migration.
+
+   If you have set per tenant limits in the Mimir configuration parameters `limits.max_global_series_per_user`, `limits.max_global_series_per_metric` via `mimir.config` or `mimir.sturcturedConfig` or via runtime overrides, double the set limits. For example:
+
+   ```yaml
+   runtimeConfig:
+     ingester_limits:
+       max_series: X # <-- double it
+     overrides:
+       tenantA:
+         max_global_series_per_metric: Y # <-- double it
+         max_global_series_per_user: Z # <-- double it
+   ```
+
+1. Create a new empty YAML file called `migrate.yaml`.
+
+1. Start the migration.
+
+   Copy the following into the `migrate.yaml` file:
+
+   ```yaml
+   ingester:
+     zoneAwareReplication:
+       enabled: true
+       migration:
+         enabled: true
+         replicas: 0
+
+   rollout_operator:
+     enabled: true
+   ```
+
+   [//]: # "ingester-step1"
+
+1. Upgrade the installation with the `helm` command and make sure to provide the flag `-f migrate.yaml` as the last flag.
+
+   In this step new zone aware statefulsets are created - but no new pods are started yet. The parameter `ingester.ring.zone_awareness_enabled: true` is set in the Mimir configuration via the `mimir.config` value. The flag `-ingester.ring.zone-awareness-enabled=false` is set on distributors, rulers and queriers. The flags `-blocks-storage.tsdb.flush-blocks-on-shutdown` and `-ingester.ring.unregister-on-shutdown` are set to true for the ingesters.
+
+1. Wait for all Mimir components to restart and be ready.
+
+1. Progressively scale zone-aware ingesters up, maximum 21 at a time.
```

Comment:
Can you explain me the 21 limit?

Reply:
I'm adding some explanation. Sublist sounds like a good idea, also the zone-aware replication config uses zone-aware: https://grafana.com/docs/mimir/latest/operators-guide/configure/configuring-zone-aware-replication/ . Going to review the text to make it consistent.

Reply:
I've reworked this part.